From 4e65607dd6f5b80dd42fb3a2dd50c38101a210d9 Mon Sep 17 00:00:00 2001
From: kszaq <kszaquitto@gmail.com>
Date: Wed, 6 Mar 2019 18:37:42 +0100
Subject: [PATCH 1/2] [AddonVideoCodec] Initialize color_primaries to
 AVCOL_PRI_UNSPECIFIED

0 is AVCOL_PRI_RESERVED0 and will trigger color conversion in renderer
which is unnecessary.

If initialized to AVCOL_PRI_UNSPECIFIED, renderer will try to guess
color primaries and in most cases will not enable color coversion.
---
 xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp
index 06fe701c6675..60eb208d3f6d 100644
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp
@@ -191,7 +191,7 @@ CDVDVideoCodec::VCReturn CAddonVideoCodec::GetPicture(VideoPicture* pVideoPictur
     pVideoPicture->iFlags = 0;
     pVideoPicture->chroma_position = 0;
     pVideoPicture->colorBits = 8;
-    pVideoPicture->color_primaries = 0;
+    pVideoPicture->color_primaries = AVColorPrimaries::AVCOL_PRI_UNSPECIFIED;
     pVideoPicture->color_range = 0;
     pVideoPicture->color_space = AVCOL_SPC_UNSPECIFIED;
     pVideoPicture->color_transfer = 0;

From e2eb1b734c5faecb2f44e8dc36e310681f547b30 Mon Sep 17 00:00:00 2001
From: kszaq <kszaquitto@gmail.com>
Date: Thu, 28 Mar 2019 09:53:18 +0100
Subject: [PATCH 2/2] [DVDVideoCodec] Properly reset color_primaries

On Reset() color_primaries should be set to AVCOL_PRI_UNSPECIFIED, not 0 which means AVCOL_PRI_RESERVED0.
---
 xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodec.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodec.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodec.cpp
index 67960fa964e9..251a674960e0 100644
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodec.cpp
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodec.cpp
@@ -39,7 +39,7 @@ void VideoPicture::Reset()
   color_space = AVCOL_SPC_UNSPECIFIED;
   color_range = 0;
   chroma_position = 0;
-  color_primaries = 0;
+  color_primaries = AVColorPrimaries::AVCOL_PRI_UNSPECIFIED;
   color_transfer = 0;
   colorBits = 8;
   stereoMode.clear();
